spec/byte_format_spec.cr

```diff
@@ -91,6 +186,73 @@ describe Crystalizer::ByteFormat do
 
   describe String do
     assert_byte_format_serialization("abc", Bytes[97, 98, 99, 0])
+
+    # Bytes corresponding to `@a = "Long description"`
+    bytes = Bytes[76, 111, 110, 103, 32, 100, 101, 115, 99, 114, 105, 112, 116, 105, 111, 110, 0]
+
+    assert_byte_format_serialization(ByteFormatTest::StringGoodRange1.new, bytes)
+    assert_byte_format_serialization(ByteFormatTest::StringGoodRange2.new, bytes)
+    assert_byte_format_serialization(ByteFormatTest::StringGoodRange3.new, bytes)
+    assert_byte_format_serialization(ByteFormatTest::StringGoodRange4.new, bytes)
+    assert_byte_format_serialization(ByteFormatTest::StringGoodRange5.new, bytes)
+    assert_byte_format_serialization(ByteFormatTest::StringGoodRange6.new, bytes)
+    assert_byte_format_serialization(ByteFormatTest::StringGoodRange7.new, bytes)
+    assert_byte_format_serialization(ByteFormatTest::StringGoodRange8.new, bytes)
+    assert_byte_format_serialization(ByteFormatTest::StringGoodRange9.new, bytes)
+
+    expect_raises Crystalizer::ByteFormat::Error, message: "String size not in range: 6..8" do
+      Crystalizer::ByteFormat.deserialize(bytes, to: ByteFormatTest::StringBadRange4)
+    end
+
+    # Bytes corresponding to `@a = "Short"`, no \0 at the end
+    bytes = Bytes[83, 104, 111, 114, 116]
+    assert_byte_format_serialization(ByteFormatTest::StringGoodSize.new, bytes)
```

Comment:
Too move between Serialization and Deserialization, because used inside the latter.